code/modules/food_and_drinks/drinks/drinks/modglass.dm

```diff
@@ -0,0 +1,173 @@
+//garnish layer defines
+#define GARNISH_RIM 1
+#define GARNISH_WEDGE 2
+#define GARNISH_SKEWER 3
+#define GARNISH_MAX 3
+
+//glass that can be reskinned via alt-click, and have garnishes added to its rim
+/obj/item/reagent_containers/food/drinks/modglass
+	name = "malleable glass"
+	desc = "Not your standard drinking glass!"
+	icon = 'icons/obj/food/modglass.dmi'
+	icon_state = "mglass-1-"
+	fill_icon = 'icons/obj/food/modglass_fillings.dmi'
+	fill_icon_thresholds = list(25,50)
+	amount_per_transfer_from_this = 10
+	volume = 50
+	custom_materials = list(/datum/material/glass=500, /datum/material/silver=100)
+	max_integrity = 30
+	spillable = TRUE
+	resistance_flags = ACID_PROOF
+	obj_flags = UNIQUE_RENAME
+	drop_sound = 'sound/items/handling/drinkglass_drop.ogg'
+	pickup_sound =  'sound/items/handling/drinkglass_pickup.ogg'
+	custom_price = 25
+	var/rim = "m"
+	var/variants = 13
+	var/list/glass_skins = list()
+	var/list/garnishes = list()
+
+/obj/item/reagent_containers/food/drinks/modglass/small
+	name = "small malleable glass"
+	icon_state = "sglass-1-"
+	custom_materials = list(/datum/material/glass=100, /datum/material/silver=100)
+	volume = 25
+	rim = "s"
+	variants = 6
+
+/obj/item/reagent_containers/food/drinks/modglass/large
+	name = "large malleable glass"
+	icon_state = "lglass-1-"
+	rim = "l"
+	variants = 5
+
+/obj/item/reagent_containers/food/drinks/modglass/Initialize()
+	for(var/x in 1 to variants)
+		glass_skins["[rim]glass-[x]-"] += icon('icons/obj/food/modglass.dmi', "[rim]glass-[x]-")
```

Comment:
dont you just want to set? why +=

Reply:
small copy+paste moment on that one, although I should  this into a global thing, since it really _really_ doesnt need to run on every single glass

Reply:
alright, replaced this shit in Initialize() with something that should be less wasteful